mpas_analysis/ocean/meridional_overturning_circulation.py

```diff
@@ -0,0 +1,445 @@
+"""
+Computation and plotting of model meridional overturning circulation.
+Will eventually support:
+  * MOC streamfunction, post-processed (currently supported)
+  * MOC streamfunction, from MOC analysis member
+  * MOC time series (max value at 24.5N), post-processed
+  * MOC time series (max value at 24.5N), from MOC analysis member
+
+Authors
+-------
+Milena Veneziani, Mark Petersen, Phillip Wolfram
+
+Last Modified
+-------------
+03/13/2017
+"""
+
+import matplotlib.pyplot as plt
+import matplotlib.colors as cols
+
+import xarray as xr
+import datetime
+import numpy as np
+import netCDF4
+import os
+
+from ..shared.constants.constants import m3ps_to_Sv
+from ..shared.plot.plotting import plot_vertical_section,\
+    timeseries_analysis_plot
+
+from ..shared.io import NameList, StreamsFile
+from ..shared.io.utility import buildConfigFullPath
+
+from ..shared.generalized_reader.generalized_reader \
+    import open_multifile_dataset
+
+from ..shared.timekeeping.utility import get_simulation_start_time, \
+    days_to_datetime
+
+from ..shared.climatology import climatology
+
+
+def moc_streamfunction(config, streamMap=None, variableMap=None):  # {{{
+    """
+    Process MOC analysis member data if available, or compute MOC at
+    post-processing if not. Plots streamfunction as well as time series
+    of max MOC at 24N.
+
+    config is an instance of MpasAnalysisConfigParser containing configuration
+    options.
+
+    If present, streamMap is a dictionary of MPAS-O stream names that map to
+    their mpas_analysis counterparts.
+
+    If present, variableMap is a dictionary of MPAS-O variable names that map
+    to their mpas_analysis counterparts.
+
+    Authors: Mark Petersen, Phillip J. Wolfram, Milena Veneziani
+    Modified: 02/23/2017
+    """
+
+    # read parameters from config file
+    inDirectory = config.get('input', 'baseDirectory')
+
+    namelistFileName = config.get('input', 'oceanNamelistFileName')
+    namelist = NameList(namelistFileName, path=inDirectory)
+
+    streamsFileName = config.get('input', 'oceanStreamsFileName')
+    streams = StreamsFile(streamsFileName, streamsdir=inDirectory)
+
+    timeSeriesStatsMonthlyAnalysisMemberFlag = namelist.get(
+        'config_am_timeseriesstatsmonthly_enable')
+    if timeSeriesStatsMonthlyAnalysisMemberFlag == '.false.':
+        raise RuntimeError('*** MPAS-Analysis relies on the existence of '
+                           'monthly\n*** averaged files: make sure to '
+                           'enable the\n*** timeSeriesStatsMonthly '
+                           'analysis member!')
```

Comment:
I'd say hold off and do it separately to keep things clearer.  There are already a lot of modified files here.